src/DevTools/mockRelayOperations.ts

```diff
@@ -0,0 +1,103 @@
+import { act, waitFor } from "@testing-library/react"
+import { OperationDescriptor } from "react-relay"
+import { MockEnvironment, MockPayloadGenerator } from "relay-test-utils"
+
+type MockedOperationResult = {
+  name: string
+  variables: any
+  operation: OperationDescriptor
+}
+
+/** extract the name and variables from a relay operation.
+ *
+ * @example
+ * ```ts
+ * const operation = await waitFor(() => env.getMostRecentOperation())
+ *
+ * expect(getOperationProps(operation)).toEqual({ name: 'MyQuery', variables: { id: '123' } })
+ * ```
+ */
+export const getOperationProps = (operation: OperationDescriptor) => {
+  const name = operation.request.node.operation.name
+  const variables = operation.request.variables
+  return { name, variables }
+}
+
+/**
+ * Asynchronously find and resolve the next operation with mock data provided
+ * in the mockResolvers format.
+ *
+ * Returns an object containing the operation, its name and variables.
+ */
+export const mockResolveMostRecentOperation = async (
+  mockRelayEnv: MockEnvironment,
+  mockResolvers: MockPayloadGenerator.MockResolvers
+): Promise<MockedOperationResult> => {
+  const operation = await waitFor(() =>
+    mockRelayEnv.mock.getMostRecentOperation()
+  )
+  await act(() => {
```

Comment:
And as @araujobarret said quite clearly, and contrary to the Bot, we don't need to to `await` act.